hs_collection_resource/views.py

```diff
@@ -40,65 +40,103 @@ def update_collection(request, shortkey, *args, **kwargs):
+            # 2) 'add': add resources in the list to collection
+            # adding a resource that is already in the collection will raise error
+            # 3) 'remove': remove resources in the list from collection
+            # removing a resource that is not in collection will raise error
+            update_type = request.POST.get("update_type", None)
+            if update_type is not None:
+                update_type = update_type.lower()
+                if update_type not in ["set", "add", "remove"]:
+                    raise Exception("Invalid value of 'update_type' parameter")
+            else:
+                # backward compatible: missing of 'update_type'
+                # results in 'set' action by default
+                update_type = 'set'
+
             if len(updated_contained_res_id_list) > len(set(updated_contained_res_id_list)):
-                raise Exception("Duplicate resources were found for adding to the collection")
+                raise Exception("Duplicate resources exist in list 'resource_id_list'")
 
             for updated_contained_res_id in updated_contained_res_id_list:
                 # avoid adding collection itself
                 if updated_contained_res_id == shortkey:
-                    raise Exception("Can not add collection itself.")
+                    raise Exception("Can not contain collection itself.")
 
             # current contained res
             res_id_list_current_collection = \
                 [res.short_id for res in collection_res_obj.resources.all()]
 
             # res to remove
             res_id_list_remove = []
-            for res_id_remove in res_id_list_current_collection:
-                if res_id_remove not in updated_contained_res_id_list:
-                    res_id_list_remove.append(res_id_remove)
-                    res_obj_remove = get_resource_by_shortkey(res_id_remove)
-                    collection_res_obj.resources.remove(res_obj_remove)
-
-                    # change "Relation" metadata in collection
-                    value = RES_LANDING_PAGE_URL_TEMPLATE.format(res_id_remove)
-                    add_or_remove_relation_metadata(add=False, target_res_obj=collection_res_obj,
-                                                    relation_type=hasPart, relation_value=value,
-                                                    set_res_modified=False)
+            if update_type == "remove":
+                res_id_list_remove = updated_contained_res_id_list
+                for res_id_remove in res_id_list_remove:
+                    if res_id_remove not in res_id_list_current_collection:
+                        raise Exception('Cannot remove resource {0} as it '
+                                        'is not currently contained '
+                                        'in collection'.format(res_id_remove))
+
+            elif update_type == "set":
```

Comment:
Not sure if the logic for 'set' is correct:

say the collection resource has currently these three as the member resources (1, 2, 3)

if the updated list comes with (1, 3, 4) with 'set' as the update_type, then the resource need to have (1, 3, 4) as the member resources. As per the code, I think it will end up having 2 member resources (1,3).